application/txhandler.go

```diff
@@ -258,39 +258,67 @@ func (tm *txHandler) UTXOGetData(txn *badger.Txn, owner *objs.Owner, dataIdx []b
 	return tm.uHdlr.GetData(txn, owner, dataIdx)
 }
 
-func (tm *txHandler) GetValueForOwner(txn *badger.Txn, owner *objs.Owner, minValue *uint256.Uint256) ([][]byte, *uint256.Uint256, error) {
-	var u [][]byte
-	v := uint256.Zero()
-	utxoIDs, value, err := tm.uHdlr.GetValueForOwner(txn, owner, minValue)
-	if err != nil {
-		utils.DebugTrace(tm.logger, err)
-		return nil, nil, err
-	}
-	v, err = v.Clone().Add(v.Clone(), value.Clone())
-	if err != nil {
-		utils.DebugTrace(tm.logger, err)
-		return nil, nil, err
-	}
-	u = append(u, utxoIDs...)
-	if v.Lt(minValue) {
-		remainder, err := new(uint256.Uint256).Sub(minValue.Clone(), v.Clone())
+func (tm *txHandler) GetValueForOwner(txn *badger.Txn, owner *objs.Owner, minValue *uint256.Uint256, pt *objs.PaginationToken) ([][]byte, *uint256.Uint256, *objs.PaginationToken, error) {
+	const maxCount = 100
+	allIds := [][]byte{}
+
+	totalValue := uint256.Zero()
+	if pt != nil {
+		var err error
+		totalValue, err = totalValue.Add(totalValue, pt.TotalValue)
 		if err != nil {
 			utils.DebugTrace(tm.logger, err)
-			return nil, nil, err
+			return nil, nil, nil, err
+		}
+	}
+
+	started := pt == nil
```

Comment:
Expand conditional for standard pattern? ( Feel free to argue more compact definition preserving single screen read-ability)

Reply:
Could you elaborate? Basically this variable is set to true as soon as we reach the last pagination type (UTXO or deposit), prevents UTXO handler from being hit if the last run already was iterating over deposits.

Reply:
I could instead do a `for ; i < length(txTypes); i++` style loop and pre-set `i` to the last paginated type (or 0 if there is no token). That however couples the logic to the iota consts even more tightly. Preference?